add RPM package spec

Wuerfelbecher opened the pull request:
Add .spec file and all that's needed to build a jellyfin.rpm for CentOS/Fedora.

brianjmurrell commented on rpm-package/jellyfin.spec:
```diff
@@ -0,0 +1,135 @@
+%global         taglib_shortcommit %(c=%{taglib_commit}; echo ${c:0:7})
+
+Name:           jellyfin
+Version:        3.5.2.git%{shortcommit}
```
This is going to lead to packages that don't successively update their predecessor when the `shortcommit` value is not lexigraphically greater than the previous release's `shortcommit`.

The proper solution here is to not use the git _sha1_ for official releases along with https://github.com/jellyfin/jellyfin/pull/418#issuecomment-451666159.

Wuerfelbecher replied:
Since i cannot use a `-` in the `Version:` tag and i see now that `sha1` is not predicable maybe we should  use something like `3.5.2.5` for git-tag `v3.5.2-5`?
```suggestion
Version:        3.5.2.5
```

Wuerfelbecher replied:
Ok changed version to 10.0.0 which would work as long as no dashes are in the git tag.

brianjmurrell commented on rpm-package/jellyfin.sudoers:
```diff
@@ -0,0 +1,19 @@
+Defaults!RESTARTSERVER_SYSTEMD !requiretty
+Defaults!STARTSERVER_SYSTEMD !requiretty
+Defaults!STOPSERVER_SYSTEMD !requiretty
+
```
I don't really have a problem with all of the above being default in an RPM install.  They are reasonable `sudo` allowances.

brianjmurrell commented on rpm-package/jellyfin.sudoers:
```diff
@@ -0,0 +1,19 @@
+
+# Uncomment to allow the server to mount iso images
+# %jellyfin ALL=(ALL) NOPASSWD: /bin/mount
+# %jellyfin ALL=(ALL) NOPASSWD: /bin/umount
```
These ones I think should be optional and require the user to enable/install.

Wuerfelbecher replied:
Emby was packaged for Redhat, but did not have a sudo policy (at least in the last couple of versions).

joshuaboniface replied:
Perfect, that should prevent the bug I saw with Debian then.

brianjmurrell commented on rpm-package/restart.sh:
```diff
@@ -0,0 +1,6 @@
+
+NAME=jellyfin
+restart_cmd="/usr/bin/systemctl restart ${NAME}"
+echo "sleep 2; sudo $restart_cmd > /dev/null 2>&1" | at now > /dev/null 2>&1
```
Why is the `sleep 2` needed?

Wuerfelbecher replied:
I reused most of the debian package also [this script](https://github.com/jellyfin/jellyfin/blob/f8a720d3d8adbdb1f092a42e592dae37ba3f25bb/debian/bin/restart.sh#L15) i assume it's to show the "server is restarting" message in the web-ui without getting killed to quickly.

joshuaboniface replied:
Yea that bit was straight outta upstream, we might eliminate it entirely soon.

Wuerfelbecher commented on rpm-package/update-db.sh:
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+db=${1:-/var/lib/jellyfin/data/library.db}
```
Your right changing the order makes more sense.

Wuerfelbecher commented:
> Should I copy this into #335?

Yes, for now i don't get a working RPM build with 10.0.0, and need to check if it's an SELinux issue or not.

Wuerfelbecher commented:
Added features of the current debian package like `-configdir` and `-logdir` to the spec also a little `%post` script that moves existing from `/var/lib/jellyfin/config` configs to `/etc/jellyfin` and symlinks it.

Also added a `build-fedora-rpm.sh` script that spits out the RPMs outside of the repo like the `buid-deb-sh` for debian. 

@joshuaboniface How does it look now, unfortunately I did not make it in time for 10.0.0.

brianjmurrell commented:
Pity this missed 10.0.1 also.  :-(

brianjmurrell commented:
Once this is landed, it would be good to get it set up in [COPR](https://copr.fedorainfracloud.org/coprs/) with webhooks to autobuild on tags and even per landing to master.  See "Integrations"

JustAMan commented on rpm-package/README.md:
```diff
@@ -0,0 +1,52 @@
+
+## ISO mounting
+
+To allow Jellyfin to mount/umonut ISO files uncomment these two lines in `/etc/sudoers.d/jellyfin-sudoers`
```
typo: `umonut` -> `umount`

Wuerfelbecher replied:
Fixed the typo.

JustAMan commented on rpm-package/README.md:
````diff
@@ -0,0 +1,52 @@
+```shell
+/usr/share/jellyfin/update-db-paths.sh <path-to-library.db> <path-to-emby-data> <path-to-jellyfin-data>
+```
+PS: Please **backup your emby database beforehand**.
````
this should not be a `PS`, it should be **the very first thing** in this section! :)

joshuaboniface replied:
Honestly, we should probably just stop trying with the whole "migrate DB" thing, since it breaks in unknown ways and just causes us a lot of headaches.

Wuerfelbecher replied:
I removed the hole DB migration part from the package since a RPM is probably not the right place for this anyway.

JustAMan commented on rpm-package/jellyfin.spec:
```diff
@@ -0,0 +1,152 @@
+export DOTNET_SKIP_FIRST_TIME_EXPERIENCE=1
+dotnet publish --configuration Release --output='%{buildroot}%{_libdir}/jellyfin' --self-contained --runtime linux-x64
+%{__install} -D -m 0644 LICENSE %{buildroot}%{_datadir}/licenses/%{name}/LICENSE
+%{__install} -D -m 0644 debian/conf/jellyfin.service.conf %{buildroot}%{_sysconfdir}/systemd/system/%{name}.service.d/override.conf
```
this doesn't feel right when `RPM` package references stuff from `debian`... if they're the same they should live in some common subdir IMHO

joshuaboniface replied:
Agreed, or just two copies - this is one of those things that should never change barring major rewrites of the packaging.

Wuerfelbecher replied:
I removed all the references to the debian package and directly referenced the source for `logging.json` as for the `override.conf` i copied the one from the debian package as it only serves as a sort of placeholder for the user to override the user under wich jellyfin runs.

JustAMan commented on rpm-package/jellyfin.spec:
```diff
@@ -0,0 +1,152 @@
+%attr(755,root,root) %{_bindir}/%{name}
+%attr(644,root,root) %{_libdir}/%{name}/*.json
+%attr(644,root,root) %{_libdir}/%{name}/*.pdb
+%attr(755,root,root) %{_libdir}/%{name}/*.dll
```
is there a reason for setting `executable` on `.dll`, `.so` and `.a` files?

Wuerfelbecher replied:
@JustAMan Now i re-encountered the problem i had originally and the reason why i set all the binaries to `755`.

For some reason has the main `jellyfin` binary `0766` permissions after the build with `dotnet` (nothing special just `dotnet publish`), so when jellyfin runs as service which is run under the `jellyfin` user it gets some permission denied errors.

JustAMan commented on rpm-package/jellyfin.spec:
```diff
@@ -0,0 +1,152 @@
+%systemd_postun_with_restart jellyfin.service
+
+%changelog
+* Tue Jan 08 2019 Thomas Büttner <thomas@vergesslicher.tech> - 10.0.0-1
```
this should be updated as it probably would land in `10.0.2` at least

joshuaboniface commented:
@Wuerfelbecher Added comments in #335 about this - if the issues @JustAMan and I raised can be solved, we can integrate the changes with #335 either as part of that commit, or afterwards by rebasing - please let us know either way.

Wuerfelbecher commented:
@joshuaboniface I addressed the issues you and @JustAMan raised.
Can @EraYaN still copy this into #335?

EraYaN commented:
I think I can just merge your branch into mine and move the files.

If that is okay with everyone I can just add your fork as a remote and merge it in. And then this one can be closed.

EraYaN commented:
I merged this into #335.
